specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json

```diff
@@ -0,0 +1,313 @@
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Settings"
+        ],
+        "operationId": "Settings_DeleteByScope",
+        "description": "Delete a setting within the given scope.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "SettingDeleteByScope": {
+            "$ref": "./examples/setting-delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeSettingParameter"
+          },
+          {
+            "$ref": "#/parameters/settingTypeParameter"
+          },
+          {
+            "$ref": "common-types.json#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded."
+          },
+          "204": {
+            "description": "OK. The request has succeeded."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Setting": {
+      "description": "Setting definition.",
+      "discriminator": "kind",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "common-types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "Specifies the kind of settings.",
+          "enum": [
+            "taginheritance"
+          ],
+          "x-ms-enum": {
+            "name": "SettingsKind",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "kind"
+      ]
+    },
+    "TagInheritanceSetting": {
+      "description": "Tag Inheritance Setting definition.",
+      "type": "object",
+      "x-ms-discriminator-value": "taginheritance",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Setting"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/TagInheritanceProperties",
+          "title": "Tag Inheritance properties"
+        }
+      }
+    },
+    "SettingsListResult": {
+      "description": "Setting list result. It contains a list of settings.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of settings.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Setting"
+          }
+        }
+      }
+    },
+    "TagInheritanceProperties": {
+      "type": "object",
+      "description": "The properties of the tag inheritance setting.",
+      "required": [
+        "preferContainerTags"
+      ],
+      "properties": {
+        "preferContainerTags": {
```

Comment:
>preferContainerTags [](http://example.com/codeflow?start=9&length=19)

Suggestion: Using enum instead of boolean is always recommended, to avoid binary nature of booleans, causing breaking changes in future, when needed to extend further.

For example:
enum preferredTags {
    Container
    ...
}

or even,
enum preferContainerTags {
    True
    False
    ...
}